extensions/notebook/src/extension.ts

```diff
@@ -231,6 +251,87 @@ async function addCell(cellType: azdata.nb.CellType): Promise<void> {
 	}
 }
 
+async function updateBookMetadata(bookPath: string): Promise<void> {
+	try {
+		// get the toc file and update the contents
+		let tocFilePath: string = path.join(bookPath, '_data', 'toc.yml');
+		let result: string;
+		fsPromises.readFile(tocFilePath).then(data => {
+			result = data.toString();
+			let contentFolders: string[] = [];
+			let headers = result.match(/- header: [a-zA-Z0-9\\.\s-]+$/gm);
```

Comment:
Where is this logic from? It would be useful to add comments explaining what all these checks are for

Reply:
so Jupyter creates a rough file with the provided list of ipynb and md files. Which needs to be formatted by the user for it to a valid book. I followed the format rules and added these. I can explain better if you have time offline.

Reply:
That's annoying, but fine. I still think that all the logic for creation should really live in the Notebook. Is there a specific reason you want it in ADS core? Being able to share the notebook and use it outside of ADS to fully create a book would be useful I'd think.

Reply:
I just added it in the core since the users cannot mess with it and I have always noticed that adding too much code in the notebook is frowned upon. But since we do have collapsible cells now, I can look into adding all the formatting on notebook side. @chlafreniere, what do you think? Is it better to have it in core or in the notebook itself?

Reply:
Removed this code and moved it to notebook.